nikola/plugins/command/plugin.py

```diff
@@ -229,104 +217,84 @@ def do_install(self, url, name, show_install_notes=True):
-        if os.path.exists(req_plug_path):
+        requirements_plugins_path = dest_path / 'requirements-plugins.txt'
+        if requirements_plugins_path.exists():
             LOGGER.info('This plugin requires other Nikola plugins.')
             LOGGER.info('Installing plugins...')
             plugin_failure = False
             try:
-                with io.open(req_plug_path, 'r', encoding='utf-8-sig') as inf:
-                    for plugname in inf.readlines():
-                        plugin_failure = self.do_install(url, plugname.strip(), show_install_notes) != 0
+                for plugin_name in requirements_plugins_path.read_text().strip().split('\n'):
+                    plugin_failure = self.do_install(url, plugin_name.strip(), show_install_notes) != 0
             except Exception:
                 plugin_failure = True
             if plugin_failure:
                 LOGGER.error('Could not install a plugin.')
                 print('Contents of the requirements-plugins.txt file:\n')
-                with io.open(req_plug_path, 'r', encoding='utf-8-sig') as fh:
-                    print(utils.indent(fh.read(), 4 * ' '))
+                print(utils.indent(requirements_plugins_path.read_text(), 4 * ' '))
                 print('You have to install those yourself manually.')
             else:
                 LOGGER.info('Dependency installation succeeded.')
 
-        confpypath = os.path.join(dest_path, 'conf.py.sample')
-        if os.path.exists(confpypath) and show_install_notes:
+        confpy_path = dest_path / 'conf.py.sample'
+        if confpy_path.exists() and show_install_notes:
             LOGGER.warning('This plugin has a sample config file.  Integrate it with yours in order to make this plugin work!')
             print('Contents of the conf.py.sample file:\n')
-            with io.open(confpypath, 'r', encoding='utf-8-sig') as fh:
-                if self.site.colorful:
-                    print(pygments.highlight(fh.read(), PythonLexer(), TerminalFormatter()))
-                else:
-                    print(fh.read())
+            if self.site.colorful:
+                print(pygments.highlight(confpy_path.read_text(), PythonLexer(), TerminalFormatter()))
+            else:
+                print(confpy_path.read_text())
         return 0
 
     def do_uninstall(self, name):
         """Uninstall a plugin."""
-        for plugin in self.site.plugin_manager.getAllPlugins():  # FIXME: this is repeated thrice
-            if name == plugin.name:  # Uninstall this one
-                p = plugin.path
-                if os.path.isdir(p):
-                    # Plugins that have a package in them need to delete parent
-                    # Issue #2356
-                    p = p + os.sep
-                    p = os.path.abspath(os.path.join(p, os.pardir))
-                else:
-                    p = os.path.dirname(p)
+        for found_name, path in self.get_plugins():
+            path: pathlib.Path
```

Comment:
Is this necessary, as `self.get_plugins`'s return value is already typed appropriately?
```suggestion
```